File: solutions/elixir/gigasecond/lib/gigasecond.ex
defmodule Gigasecond do
  @moduledoc false
  @type erlangDateTime ::
          {{pos_integer, pos_integer, pos_integer}, {pos_integer, pos_integer, pos_integer}}

  @doc """
  Calculate a date one billion seconds after an input date.
  """
  @spec from(erlangDateTime) :: erlangDateTime
  def from({{year, month, day}, {hours, minutes, seconds}}) do
    NaiveDateTime.new!(year, month, day, hours, minutes, seconds)
    |> NaiveDateTime.add(1_000_000_000)
    |> NaiveDateTime.to_erl()
  end
end

File: solutions/go/gigasecond/gigasecond.go
// Package gigasecond is for working with units of one thousand million seconds.
package gigasecond

import "time"

// AddGigasecond returns the date and time one gigasecond after a certain date.
func AddGigasecond(t time.Time) time.Time {
	return t.Add(time.Second * 1_000_000_000)
}
